daemon/container.go

```diff
@@ -1627,3 +1618,93 @@ func (container *Container) monitorExec(execConfig *execConfig, callback execdri
 
 	return err
 }
+
+type archiveFilter struct {
+	basePath string
+	files    []string
+}
+
+func archiveGlob(basePath string) (*archiveFilter, error) {
+	paths, err := filepath.Glob(basePath)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(paths) == 0 {
+		return nil, os.ErrNotExist
+	}
+
+	baseDir := findSignificantParent(paths)
+
+	relative := make([]string, len(paths))
+
+	for i, f := range paths {
+		relative[i] = f[len(baseDir):]
+	}
+
+	return &archiveFilter{baseDir, relative}, nil
+}
+
+func findSignificantParent(paths []string) string {
```

Comment:
This seems like a lot of work.  Would something a little smaller like this work?  https://gist.github.com/duglin/c88c3d5fde9c9ac6fd22